### DISCUSSION TITLE
Civitai cannot resolve sui_model hashes?

### DISCUSSION THREAD

**Discussion by azmeenafandi** [mcmonkeyprojects/SwarmUI#493] [Q&A] [answered]
Greetings everyone!

I could not get civitai to recognize the loras and embeddings used to generate images. A sample sui_model data is as follows:

```
  "sui_models": [
    {
      "name": "SDXL/realismByStableYogi_v5XLFP32.safetensors",
      "param": "model",
      "hash": "0x71ff9d7ffa143818519508f9afd9968574f61126b243e4bcd30f7762b3fedea1"
    },
    {
      "name": "positive/Stable_Yogis_PDXL_Positives.safetensors",
      "param": "used_embeddings",
      "hash": "0xec06a54a360498de7ded054c5cb01c9b77cc5507aabbe8b76ded4ad5ad5431e9"
    },
    {
      "name": "negative/Stable_Yogis_PDXL_Negatives-neg.safetensors",
      "param": "used_embeddings",
      "hash": "0xa6c5f1fcf5f7f8877391b98ee9d874f7e52946e2dd9fa280771ebb16ad1204f1"
    },
    {
      "name": "negative/SDXL_Negative_-_XL_NEG.safetensors",
      "param": "used_embeddings",
      "hash": "0xb469afaf9b37b8c4332197dda724b726fd6ed19b519d3b608189ca8e55dec9de"
    },
    {
      "name": "negative/Negative_Hands_-_v0-1-XL.safetensors",
      "param": "used_embeddings",
      "hash": "0x67a0b9310462e58b27527736738b053b71c2ec7cbc94a943e3ea4aa623363608"
    },
    {
      "name": "pony/Consistency Corrector XL.safetensors",
      "param": "loras",
      "hash": "0x753c1eb1510ed189ae7d2c513107d8291a96c0480a5a0ef91e0c44c0df9d1101"
    },
    {
      "name": "pony/Expressive_H-000001.safetensors",
      "param": "loras",
      "hash": "0x5671f20a9a6ba68a857c44041315440f64e56094e87f38adb0d369a863b527c6"
    },
    {
      "name": "pony/RealisticSkinv1_ponyv6_loraplus.safetensors",
      "param": "loras",
      "hash": "0x4ad0c6c0e8525016a56dd4aee82d72e58955d1a0d7e8d99b920bfeb4ee0646a6"
    }
  ]
```

I noticed that the hash values doesn't match any of the hash values listed on civitai, which is probably why they are not autolinked in image posts.

For example, here are the hash values listed for Consistency Corrector XL on civitai:
AUTOV1: 5D41DC69
AUTOV2: 9ED164F449
AUTOV3: 753C1EB1510E
SHA256: 9ED164F449688DC4B8152E9C0B38D5A9A41A7AF21D49635BD6957E149DB68724
CRC32: 6794BFAF
BLAKE3: 4D2C802A2EC906F73A93AF34D68762B966E6BD43536668DAF2B4804CF8C10C65
 
None of the above matches SwarmUI's generated hash: 0x753c1eb1510ed189ae7d2c513107d8291a96c0480a5a0ef91e0c44c0df9d1101

Is there anything I can do to help fix this?

Thanks and regards

**Comment by mcmonkey4eva** [comment, marked as answer]
The "AutoV3" is the match.
```
753C1EB1510E - Civit autov3
753c1eb1510ed189ae7d2c513107d8291a96c0480a5a0ef91e0c44c0df9d1101 - swarm
```
Civit just cuts it short. Swarm's metadata lookup automatically cuts it short when making calls it civit.

Regarding civitai not reading the values, that's something you'd have to prod civit about

**Reply by azmeenafandi** [reply]
Ahhh... I see. Thank you for answering, much appreciated!

**Reply by azmeenafandi** [reply]
Hi @mcmonkey4eva,

I dug around a bit more and noticed that GetOrGenerateTensorHashSha256 always prefixes hashes with "0x". These hashes are then generated and embedded under sui_models. I decided to just remove this prefix and recompile the project locally. I'm thinking that this is probably why civitai didn't recognize the hashes.

However for the life of me, after hours of reading and performing dotnet clean, dotnet restore and dotnet build; I simply cannot get SwarmUI to implement this change. It's like I didn't even perform the edit.

As you can obviously tell, I'm not a dotnet guy even in the slightest.

I respect that you won't implement any unnecessary changes to your project. I hope that you can nudge me in the right direction on how to recompile SwarmUI on my local computer to test out this slight change.

Thank you.

**Reply by mcmonkey4eva** [reply]
don't randomly hack around source code, that's not how that works. Again, you'd have to bug civit to add support for processing those hashes, no edits to swarm will change anything

**Comment by kalebbroo** [comment]
To answer your other question, there is a dev PowerShell script to run when working in the files. You have to start Swarm using that to get it to recognize your changes.
